mlflow/pyfunc/__init__.py

```diff
@@ -1999,6 +2009,25 @@ def predict(model_input: List[str]) -> List[str]:
                 python_model, input_arg_index, input_example=input_example
             ):
                 mlflow_model.signature = signature
+        elif isinstance(python_model, ChatModel):
```

Comment:
Can we do any validation/warning if customer specifies custom signature with ChatModel? (better than failing at runtime)

Reply:
oh yes that's true, i'll throw a warning to say that the signature will be overridden and that it must conform to the spec

Reply:
woah actually this brought up a bug in my implementation—if the user specifies a signature, the model actually doesn't get saved as a ChatModel due to the `elif` in line 2005 above. i guess it's `elif` because this block contains a lot of validation/signature inference logic that we can skip if the user provides the signature themself. however, for `ChatModel` we always want to do these validations (e.g. output validation)

cc @B-Step62 what do you think about raising an exception when trying to save a ChatModel subclass with a signature, e.g:

```python
if signature is not None:
  if isinstance(python_model, ChatModel):
    raise MlflowException("ChatModel subclasses specify a signature automatically, please remove the provided signature from the log_model() or save_model() call.")
  mlflow_model.singature = signature
elif python_model is not None:
  # no change from this PR
```

another way is making a separate block for ChatModels, e.g:
```python
if isinstance(python_model, ChatModel):
  # move ChatModel logic to this block
  ...
elif signature is not None:
  # no change
  ...
elif python_model is not None:
  # no change
  ...
```

Reply:
Nice finding! I agree with throwing. Warning on happy path can be easily overlooked and almost invisible in automated environment.